matchzoo/models/drmmtks_model.py

```diff
@@ -0,0 +1,162 @@
+"""An implementation of DRMMTKS Model."""
+from matchzoo import engine
+import typing
+import numpy as np
+import keras.backend as K
+from keras.models import Model
+from keras.layers import Input, Embedding, Dense, Lambda, Activation, \
+    Dot, Flatten
+from keras.activations import softmax
+
+def show_tensor_info(name, input):
+    print('[Layer]: %s\t[Shape]: %s\n' % (name, input.get_shape().as_list()))
+
+class DRMMTKSModel(engine.BaseModel):
+    """
+    DRMMTKS Model.
+
+    Examples:
+        >>> model = DRMMTKSModel()
+        >>> model.guess_and_fill_missing_params()
+        >>> model.build()
+
+    """
+
+    @classmethod
+    def get_default_params(cls) -> engine.ParamTable:
+        """:return: model default parameters."""
+        params = super().get_default_params()
+        params['optimizer'] = 'adam'
+        params['input_shapes'] = [(5,), (300,)]
+        params.add(engine.Param('trainable_embedding', False))
+        params.add(engine.Param('embedding_dim', 300))
+        params.add(engine.Param('vocab_size', 100))
+        params.add(engine.Param('top_k', 10))
+        params.add(engine.Param('hidden_sizes', [5, 1]))
+        params.add(engine.Param('embedding_mat', None))
+        params.add(engine.Param('embedding_random_scale', 0.2))
+        return params
+
+    @property
+    def embedding_mat(self) -> np.ndarray:
+        """Get pretrained embedding for ArcI model."""
+        # Check if provided embedding matrix
+        if self._params['embedding_mat'] is None:
+            s = self._params['embedding_random_scale']
+            self._params['embedding_mat'] = \
```

Comment:
Do not put values like this into `self._params`. Remember that `self._params` should be able to deterministically re-create a model, and this will potentially cause problems when saving.

Reply:
If you want to `remember` an embedding matrix, use something like a `self._embedding_mat` instead.

Reply:
I agree with you to move the `_embedding_mat` out of the `_params`.